src/test/java/com/comphenix/protocol/PacketTypeTest.java

```diff
@@ -52,60 +51,28 @@ public static void beforeClass() {
 
     @AfterAll
     public static void afterClass() {
-        PacketType.onDynamicCreate = __ -> {
+        PacketType.onDynamicCreate = (x, y) -> {
         };
     }
 
-    @SuppressWarnings("unchecked")
-    // @Test
-    // public static void main(String[] args) throws Exception {
-    public void generateNewPackets() throws Exception {
-        MinecraftReflectionTestUtil.init();
-
-        Set<Class<?>> allTypes = new HashSet<>();
-        List<Class<?>> newTypes = new ArrayList<>();
-
-        EnumProtocol[] protocols = EnumProtocol.values();
-        for (EnumProtocol protocol : protocols) {
-            System.out.println(WordUtils.capitalize(protocol.name().toLowerCase()));
-
-            Field field = EnumProtocol.class.getDeclaredField("k");
-            field.setAccessible(true);
-
-            Map<EnumProtocolDirection, Object> map = (Map<EnumProtocolDirection, Object>) field.get(protocol);
-            for (Entry<EnumProtocolDirection, Object> entry : map.entrySet()) {
-                Field mapField = entry.getValue().getClass().getDeclaredField("b");
-                mapField.setAccessible(true);
-
-                Map<Class<?>, Integer> reverseMap = (Map<Class<?>, Integer>) mapField.get(entry.getValue());
-
-                Map<Integer, Class<?>> treeMap = new TreeMap<>();
-                for (Entry<Class<?>, Integer> entry1 : reverseMap.entrySet()) {
-                    treeMap.put(entry1.getValue(), entry1.getKey());
-                }
-
-                System.out.println("  " + entry.getKey());
-                for (Entry<Integer, Class<?>> entry1 : treeMap.entrySet()) {
-                    System.out.println(generateNewType(entry1.getKey(), entry1.getValue()));
-                    allTypes.add(entry1.getValue());
+     public static void main(String[] args) throws Exception {
+//    public void generateNewPackets() throws Exception {
+        BukkitInitialization.initializeAll();
 
-                    try {
-                        PacketType.fromClass(entry1.getValue());
-                    } catch (Exception ex) {
-                        newTypes.add(entry1.getValue());
-                    }
-                }
-            }
-        }
+        PacketType.onDynamicCreate = (type, className) -> System.out.printf("%s, %s = new PacketType(PROTOCOL, SENDER, %s, \"%s\")\n", type.getProtocol(), type.getSender(), formatHex(type.getCurrentId()), className);
```

Comment:
love it

Reply:
Seemed like the easiest way to me I would also like to parse the packet names and only print the short name if possible:
e.g. `ServerboundClientInformationPacket` -> `ClientInformation`